### DISCUSSION TITLE
Project proposal template

### DISCUSSION THREAD

**Discussion by jawache** [Green-Software-Foundation/oc#7] [General]
Currently we have a bit of a miss mash of a process that's evolved over time to propose projects. 

There was a period of time in the early days of the GSF when the number of project ideas far surpassed our ability to support so we set the bar high for new project proposals. At least 3 people and 2 organizations had to be onboard with an idea before we launched it officially. That eventually evolved to a PR-FAQ based process documented here: https://github.com/Green-Software-Foundation/pr-faqs.

The process as described in the OC project lifecycle doc has a lower bar for incubating a project which allows for more experimentations, projects can be launched in an incubation mode even if it's just one person with an idea.

We need a project proposal template, our current repo [doesn't have one](https://github.com/Green-Software-Foundation/oc/blob/main/project-lifecycle.md#proposal), this is the one from Hyperledger https://hyperledger.github.io/hyperledger-hip/, we also have a PR-FAQ process which has some pros and cons (One con seem to be writing a PR-FAQ is more of an art form => increases the barrier to proposing projects)

I'd like to propose something in the middle, more of a form with fields people have to fill out, but with sections that pull out the best parts of the PR-FAQ process.

I propose that people submit projects by creating a Pull Request into **THIS REPO**, according to the template below. This committee then discusses the proposal and votes as to whether to accept it into incubation

Proposed Fields:

- **Working Title**: (Official names for projects are very important, we should have a naming process that involves perhaps even our writers/comms people on staff)
- **Tagline**: A very short description of the project, **MAX 8 WORDS**
- **Abstract**: A summary of the project, **MAX 50 WORDS**
- **Audience**: Who is the primary audience for this solution? Who would we need to market this product to for it to be a success at scale? Where do these people go for their information? Which conferences would people talk about this project in? **MAX 100 WORDS**
- **ToC**: How will this project support one or more of our ToC pillars
   - e.g. It will improve **tooling** by “xxxxx”.
- **Problem**: List 3-5 problems this audience has that this project will attempt to address. Describe Why is this project better at solving a problem compared to parallel proposals or implemented projects? **MAX 200 WORDS**
- **Solution**: Try to make this as detailed as possible. The topics given below are just suggestions; address them only if they are relevant to your problem:
   - Rough design and scenarios on the probable effects, if any.
   - The use of diagrams is encouraged to elucidate concepts.
   - Address any possible objections.
   - Explain how the solution solves each of the problems listed in the problem section.
- **Closure**: How do we know that the project succeeded? This has to be measurable if possible. Make references to successor projects, if any.
- **FAQ**: Add anything here that doesn't fit into the above sections. It can be blank, to begin with; as questions are asked and points clarified use this section to document those clarifications.

The points above take the best parts from the PR-FAQ process. Keeping it to one page forced summarization which is a useful exercise to surface the key important messages related to a project, but also a strong focus on problem -> solution -> audience is also something I really like about the PR-FAQ process. I've also added a section to describe how this project will support one of our ToC pillars, all our work really should be driving one or more of those pillars from now on.

**Comment by Sealjay** [comment]
I like this process, but could we keep pr-faq as an optional document? Maybe a markdown they could complete.

I think some people like to complete this as it helps to direct their vision.

Or at least I like the "headline" and "made up quote from a business leader" approach, as it sometimes elicits feedback on why the project will actually solve a business imperative.

Are we going to use a google doc, or keep the hyperledger process of forking a repo? I'm ok with forking as it keeps it public.

**Comment by jawache** [comment]
- Add a field to the form for "made up quote from a business leader"
- Process is to create a file in the OC repo and a PR which is reviewed on the OC call.

**Reply by Sealjay** [reply]
So rather than this:
- https://github.com/hyperledger/hyperledger-hip/
    - Fork this repository.
    - Fill out the Proposal Template and save it into the HIPs subdirectory under the name of your project, such as mynewproject.md.

We would instead ask someone to complete a pull request into the OC project, into a proposals directory, copying a file template.
